src/pages/index.js

```diff
@@ -0,0 +1,369 @@
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Delete", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Delete", "Deleting...");
+    });
+}
+
+function handleLike(evt, id) {
+  //1. check whether card is currently liked or not
+  //2. call the changeLike methods, passing it to the appropriate arguments
+  //3. handle the response (.then and .catch)
+  //4. in the .then, toggle active class
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+
+  api
+    .changeLike(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-btn_liked");
+    })
+    .catch(console.error);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameElement = cardElement.querySelector(".card__title");
+  const cardImageElement = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  //TODO - if the card is liked, set the active class to the card
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.setAttribute("alt", data.name);
+
+  cardLikeBtn.addEventListener("click", (evt) =>
+    //cardLikeBtn.classList.toggle("card__like-btn_liked");
+    handleLike(evt, data._id)
+  );
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  cardImageElement.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaption.textContent = data.name;
+    previewModalImage.src = data.link;
+    previewModalImage.setAttribute("alt", data.name);
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+  modal.addEventListener("mousedown", handleOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+  modal.removeEventListener("mousedown", handleOverlay);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  //submitBtn.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+
+      closeModal(editModal);
+      editFormElement.reset();
+      disableButton(submitBtn, false);
```

Comment:
`disableButton(submitBtn, false);`


`disableButton` requires `settings` rather than `false` or `true`


```js
 disableButton(submitBtn, settings);
```